src/httprpc.cpp

```diff
@@ -314,9 +314,14 @@ static bool InitRPCAuthentication()
         LogPrintf("Config options rpcuser and rpcpassword will soon be deprecated. Locally-run instances may remove rpcuser to use cookie-based auth, or may be replaced with rpcauth. Please see share/rpcauth for rpcauth auth generation.\n");
         strRPCUserColonPass = gArgs.GetArg("-rpcuser", "") + ":" + gArgs.GetArg("-rpcpassword", "");
     }
-    if (gArgs.GetArg("-rpcauth", "") != "") {
+
+    if (!gArgs.GetArgs("-rpcauth").empty()) {
         LogPrintf("Using rpcauth authentication.\n");
         for (const std::string& rpcauth : gArgs.GetArgs("-rpcauth")) {
+            if (rpcauth.empty()) {
+                LogError("Invalid (empty) -rpcauth argument\n");
+                return false;
+            }
```

Comment:
In commit "fix: increase consistency of rpcauth parsing" (dc38788312e7cdbbc106103e671d548d5c236862)

I think it would be better for code clarity to drop these 4 lines. The only thing they are doing is triggering a special-case "Invalid (empty) -rpcauth argument" error instead of the more generic "Invalid -rpcauth argument." error that would already happen below. The generic error seems sufficient, and If it is not sufficient it would still seem better to handle the two errors together rather than separately.

Reply:
Thanks.  Will do (i.e. handle the errors together)

Reply:
Thanks.  Dropped the four lines so the errors are handled together.  Reordered the commits so the tests are first, then the fix.  The tests were adjusted to prevent "test each commit" from failing.